### PR TITLE
feat: handle subscription created webhook on import

### PR DESCRIPTION
**Pull request by asbiin**
The idea is to handle the [subscription_created](https://developer.paddle.com/webhook-reference/553f846002eee-subscription-created) event.
It's already handled by [`WebhookController.php`](https://github.com/laravel/cashier-paddle/blob/1.x/src/Http/Controllers/WebhookController.php), but only it the user already exists and initiated the transaction.

Here I want to handle the event for the migration/import process.

### CODE REVIEW

File: app/Listeners/WebhookReceivedListener.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     * @return void
+     *
+     * @throws \Laravel\Paddle\Exceptions\InvalidPassthroughPayload
+     */
+    protected function handleSubscriptionCreated(array $payload)
+    {
+        $customer = $this->findOrCreateCustomer($payload, $payload['passthrough']);
+
+        $trialEndsAt = $payload['status'] === Subscription::STATUS_TRIALING
+            ? Carbon::createFromFormat('Y-m-d', $payload['next_bill_date'], 'UTC')->startOfDay()
```
Comment (djaiss) [inline app/Listeners/WebhookReceivedListener.php:55]: By default it's UTC and by default it's also start of day, but at least we are sure.

File: app/Listeners/WebhookReceivedListener.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+     *
+     * @throws \Laravel\Paddle\Exceptions\InvalidPassthroughPayload
+     */
+    protected function findOrCreateCustomer(array $payload, ?string $passthrough): User
+    {
+        if (! empty($passthrough)) {
+            $passthrough = json_decode($passthrough, true);
+        }
+
+        if (is_array($passthrough) && isset($passthrough['billable_id'], $passthrough['billable_type'])) {
```
Comment (djaiss) [inline app/Listeners/WebhookReceivedListener.php:88]: The method takes a string so the `is_array` method will always return false, I think.

File: app/Listeners/WebhookReceivedListener.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            // It will be handled by cashier's WebhookController
+            throw new InvalidPassthroughPayload();
+        }
+
+        return tap(User::firstOrCreate(
+            ['email' => $payload['email']],
+            ['name' => $payload['email']]
```
Comment (djaiss) [inline app/Listeners/WebhookReceivedListener.php:95]: Are you sure? `name` => `email` ?

File: app/Listeners/WebhookReceivedListener.php (hunk 1 above)
Comment (asbiin) [reply app/Listeners/WebhookReceivedListener.php:55]: See https://github.com/laravel/cashier-paddle/blob/198b40a790acad4a1b0ace3f6dab3794bac0315c/src/Http/Controllers/WebhookController.php#L233

File: app/Listeners/WebhookReceivedListener.php (hunk 3 above)
Comment (asbiin) [reply app/Listeners/WebhookReceivedListener.php:95]: Yes, we don't have the name in this scenario.

File: app/Listeners/WebhookReceivedListener.php (hunk 2 above)
Comment (asbiin) [reply app/Listeners/WebhookReceivedListener.php:88]: The variable is changed 2 lines below

Comment (asbiin) [reply app/Listeners/WebhookReceivedListener.php:88]: BTW the code is similar to https://github.com/laravel/cashier-paddle/blob/198b40a790acad4a1b0ace3f6dab3794bac0315c/src/Http/Controllers/WebhookController.php#L260

**Review by djaiss — APPROVED** [review]
Superb.

**Pull request merged**